### PR TITLE
[Java] BAS OnPrem connectivity instructions

### PR DESCRIPTION
**Pull request by Johannes-Schneider**
## What Has Changed?

This PR adds a small section about how to setup your development environment in BAS when OnPrem connectivity is needed.

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
- [x] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.

### CODE REVIEW

File: docs/java/features/connectivity/destinations.mdx [hunk 1]
```diff
@@ -468,14 +468,36 @@ See the example value above for reference.
 
 #### Destinations With Proxy Type `OnPremise`
 
-We have to some perform configuration steps to tell the SAP Cloud SDK how to access the SAP BTP destination service and connectivity service.
+We have to perform some configuration steps to tell the SAP Cloud SDK how to access the SAP BTP destination service and connectivity service.
```
Comment (artemkovalyov) [inline docs/java/features/connectivity/destinations.mdx:471]:
```suggestion
To access the **Destination** and **Connectivity** service on SAP BTP we have to configure the SAP Cloud SDK accordingly.
```

File: docs/java/features/connectivity/destinations.mdx [hunk 2]
```diff
@@ -468,14 +468,36 @@ See the example value above for reference.
...
 
 :::note Are your client and server located on the same Network?
 If your localhost (client) resides in the same network zone as the target system (server) you can access it directly.
 Configuration of SAP BTP connectivity services is not needed.
 All you have to do is creating a destination via environment variable `destinations`.
 :::
 
+##### Developing in Business Application Studio
+
+If you are developing your application in Business Application Studio (BAS), reaching On-Premise systems can be achieved by performing the following steps:
```
Comment (artemkovalyov) [inline docs/java/features/connectivity/destinations.mdx:481]:
```suggestion
If you are developing an application in the **Business Application Studio (BAS)**, and want to reach On-Premise systems, perform the following steps:
```

File: docs/java/features/connectivity/destinations.mdx [hunk 3]
```diff
@@ -468,14 +468,36 @@ See the example value above for reference.
...
+
+1. Make sure your BAS instance is part of the same CF space as your Cloud Connector.
+1. Add two properties called `WebIDEEnabled` and `HTML5.DynamicDestination`, both with value `true`, to your On-Premise destination.
```
Comment (artemkovalyov) [inline docs/java/features/connectivity/destinations.mdx:484]:
```suggestion
1. Add properties `WebIDEEnabled` and `HTML5.DynamicDestination` to your On-Premise destination with the value `true`.
```

Comment (artemkovalyov) [reply docs/java/features/connectivity/destinations.mdx:484]: Maybe having an example here?

File: docs/java/features/connectivity/destinations.mdx [hunk 4]
```diff
@@ -468,14 +468,36 @@ See the example value above for reference.
...
+1. In BAS, configure your `destinations` environment variable as follows:
```
Comment (artemkovalyov) [inline docs/java/features/connectivity/destinations.mdx:485]:
```suggestion
1. In BAS, configure the environment variable of every On-Premise `destination` as follows:
```

Comment (artemkovalyov) [reply docs/java/features/connectivity/destinations.mdx:485]: I was confused between plural, possessive case and what exactly relates to what. Check if my suggestion sound like correct for you.

File: docs/java/features/connectivity/destinations.mdx [hunk 5]
```diff
@@ -468,14 +468,36 @@ See the example value above for reference.
...
+    "type": "HTTP",
+    "name": "<DESTINATION-NAME>",
+    "url": "http://<DESTINATION-NAME>.dest",
+    "ProxyHost": "localhost",
+    "ProxyPort": 8887
+  }
+]
+```
+
+Please refer to [the BAS connectivity guide](../../../js/guides/bas-external-system) for more information and a detailed description of the technical background.
```
Comment (artemkovalyov) [inline docs/java/features/connectivity/destinations.mdx:499]:
```suggestion
Please refer to [the BAS connectivity guide](../../../js/guides/bas-external-system) created for the JavaScript SDK to get more information and a detailed description of the technical background. The information there equally applies to the Java SDK.
```

**Review by artemkovalyov — APPROVED** [review]
I think it's well done. I have a clarity-related suggestion. What to apply is at your discretion.

File: docs/java/features/connectivity/destinations.mdx (hunk 3 above)
Comment (Johannes-Schneider) [reply docs/java/features/connectivity/destinations.mdx:484]: You mean like a screenshot?

File: docs/java/features/connectivity/destinations.mdx (hunk 4 above)
Comment (Johannes-Schneider) [reply docs/java/features/connectivity/destinations.mdx:485]: No, that is actually not correct. The SDK looks for one specific environment variable, which is called `destinations`. This variable contains multiple service bindings. The example I provide, however, displays the entire `destinations` variable... So I would actually leave the description as is.

**Pull request merged**